sdk/include/reactos/shellutils.h

```diff
@@ -597,21 +597,14 @@ void DumpIdList(LPCITEMIDLIST pcidl)
     DbgPrint("End IDList Dump.\n");
 }
 
-struct CCoInit
+template<HRESULT(WINAPI*InitFunc)(void*), void(WINAPI*UninitFunc)()> struct CCoInitBase
 {
-    CCoInit()
-    {
-        hr = CoInitialize(NULL);
-    }
-    ~CCoInit()
-    {
-        if (SUCCEEDED(hr))
-        {
-            CoUninitialize();
-        }
-    }
     HRESULT hr;
+    CCoInitBase() : hr(InitFunc(NULL)) { }
+    ~CCoInitBase() { if (SUCCEEDED(hr)) UninitFunc(); }
```

Comment:
https://reactos.org/wiki/Coding_Style#Line_breaking
```suggestion
    ~CCoInitBase()
    {
        if (SUCCEEDED(hr))
            UninitFunc();
    }
```
and makes step-by-step debugging easier.

Reply:
I can promise you that nobody will ever need to step-by-step debug this but OK. (You can also temporarily turn off source debugging if you really needed to step here)